app/services/DelegationService.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import connectors.DelegationConnector
+import models.DelegationModel
+import play.api.Logger
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+trait DelegationService {
+
+  val delegationConnector: DelegationConnector
+
+  def delegationCall(id: String)(implicit hc: HeaderCarrier): Future[Option[DelegationModel]] = {
+    delegationConnector.delegationDataCall(id).map { response =>
+      response.status match {
+        case _ => response.json.asOpt[DelegationModel]
+      }
+    }.recover {
+      case e =>
+        Logger.warn(s"[DelegationService][delegationCall] no Delegation Returned")
```

Comment:
AMBER - the little s before the string doesn't need to be there, alternatively, keep it and add the exception 'e' to the log.

Reply:
removed